python/tank/errors.py

```diff
@@ -37,3 +37,18 @@ def __init__(self):
         """
         super(TankErrorProjectIsSetup, self).__init__("You are trying to set up a project which has already been set up. "
                                                       "If you want to do this, make sure to set the force parameter.")
+
+class TankAuthenticationError(TankError):
+    """
+    Exception that indicates an error has occurred.
```

Comment:
This description is too generic! 
Also, why do we need a specific TankAuth error? Can't we just use errors raised by the authentication module? And if we had to do a tank exception type, should it derive from one of the auth module exceptions rather than from tank error?